lms/static/scripts/frontend_apps/components/Table.js

```diff
@@ -24,8 +29,34 @@ function nextItem(items, currentItem, step) {
   return items[index + step];
 }
 
+/**
+ * @typedef TableColumn
+ * @prop {string} label - Header label for the column
+ * @prop {string} className - Additional classes for the column's `<th>` element
+ */
+
+/**
+ * @template Item
+ * @typedef TableProps
+ * @prop {string} accessibleLabel - An accessible label for the table.
+ * @prop {TableColumn[]} columns - The columns to display in this table.
+ * @prop {Item[]} items - The items to display in this table.
```

Comment:
I find the name `items` a little confusing. Are we talking about cell values here? Does a single `item` contain a row's worth of information? (I went down and read the code; it seems so).

Reply:
The assumption here is that the table is displaying a list of items (eg. files, book chapters, books) and information from those items is somehow mapped to columns. `renderItem` defines the mapping from item to row contents.